R/SSplotTags.R

```diff
@@ -43,380 +43,399 @@
+           "Observed and expected post-latency tag recaptures by year and tag group", #8
+           "Summarized observed and expected numbers of recaptures by fleet", #9
+           "Pearson residuals by tag group"), #10 
+           plotdir="default",
+           verbose=TRUE)
+{
+  # subfunction to write png files
+  pngfun <- function(file, caption=NA){
+    png(filename=file.path(plotdir, file),
+        width=pwidth, height=pheight, units=punits, res=res, pointsize=ptsize)
+    plotinfo <- rbind(plotinfo, data.frame(file=file, caption=caption))
+    return(plotinfo)
+  }
+  plotinfo <- NULL
 
-    tagdbase2 <- replist[["tagdbase2"]]
-    if (is.null(tagdbase2) || nrow(tagdbase2) == 0) {
-      if (verbose) {
-        message("skipping tag plots because there's no tagging data")
-      }
-    } else {
-      # filter tag groups if requested
-      if (!is.null(taggroups)) {
-        tagdbase2 <- tagdbase2[tagdbase2[["Repl."]] %in% taggroups, ]
-        message(
-          "Filtered tag groups for plotting based on input vector taggroups\n",
-          "Plots will show", length(unique(tagdbase2[["Repl."]])),
-          "out of", length(unique(replist[["tagdbase2"]][["Repl."]])),
-          "total included in the model."
-        )
-      }
+  if(plotdir=="default") plotdir <- replist$inputs$dir
 
-      # calculations needed for printing to multiple PNG files
-      grouprange <- unique(tagdbase2[["Repl."]])
-      ngroups <- length(unique(tagdbase2[["Repl."]]))
-      npages <- ceiling(ngroups / (tagrows * tagcols))
-      nseasons <- replist[["nseasons"]]
-      width <- 0.5 / nseasons
-      tagreportrates <- replist[["tagreportrates"]]
-      tagrelease <- replist[["tagrelease"]]
-      tagsalive <- replist[["tagsalive"]]
-      tagtotrecap <- replist[["tagtotrecap"]]
-      if (is.null(latency)) {
-        latency <- replist[["tagfirstperiod"]]
-      }
+  tagdbase2 <- replist$tagdbase2
+  if(is.null(tagdbase2) || nrow(tagdbase2)==0){
+    if(verbose){
+      message("skipping tag plots because there's no tagging data")
+    }
+  }else{
+    # filter tag groups if requested
+    if(!is.null(taggroups)){
+      tagdbase2 <- tagdbase2[tagdbase2$Rep %in% taggroups,]
+      message("Filtered tag groups for plotting based on input vector taggroups\n",
+              "Plots will show", length(unique(tagdbase2$Rep)),
+              "out of", length(unique(replist$tagdbase2$Rep)),
+              "total included in the model.")
+    }
 
-      tagfun1 <- function(ipage = 0) {
-        # obs & exp recaps by tag group
-        par(
-          mfcol = c(tagrows, tagcols), mar = c(2.5, 2.5, 2, 1),
-          cex.main = cex.main, oma = c(2, 2, 2, 0)
-        )
-        if (npages > 1 & ipage != 0) {
-          grouprange <- intersect(
-            grouprange,
-            1:(tagrows * tagcols) + tagrows * tagcols * (ipage - 1)
-          )
+    # calculations needed for printing to multiple PNG files
+    grouprange     <- unique(tagdbase2$Rep)
```

Comment:
I will work on this now and push a commit once I am finished.

Reply:
I left the dollar signs for now (but I can always change if you do think it would be useful) but committed changing `tagdbase2$Rep` to `tagdbase2[["Rep1."]]`
(see commit https://github.com/r4ss/r4ss/pull/455/commits/d74d2e3cffaf869e9452a54b50d5cc666b5761b3)

Reply:
Perfect! I ran a function we have in r4ss called `rm_dollar_signs` to change the rest of them.